packages/databases-collections-list/src/items-grid.tsx

```diff
@@ -86,53 +79,9 @@ type ItemsGridProps<T> = {
   renderItem: RenderItem<T>;
 };
 
-const CONTROLS_HEIGHT = (spacing[5] as number) + 36;
-
-const pushRight = css({
-  marginLeft: 'auto',
-});
-
-// We use this context to pass components that are aware of outer state of the
-// v-list to the list header. This is needed so that we can define this outer
-// component outside of the list component scope and avoid constant re-mounts
-// when component constructor is re-created. We do this so that controls can be
-// part of the list header and scroll up when the list is scrolled
```

Comment:
This is still relevant, if you want to pass new data to controls component, it needs to come through context and you can't pass a function that is always recreated on render to `renderHeader` method

Reply:
Resolving this as i undid this change.